### PR TITLE
Support feed item images

### PR DESCRIPTION
**Pull request by gmemstr**
After some work from myself and florentheobin we've gotten image support for feeds fully done, including tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
-## gorilla/feeds
-[![GoDoc](https://godoc.org/github.com/gorilla/feeds?status.svg)](https://godoc.org/github.com/gorilla/feeds) [![Build Status](https://travis-ci.org/gorilla/feeds.png?branch=master)](https://travis-ci.org/gorilla/feeds)
+## gmemstr/feeds
```
Comment (kisielk) [inline README.md]: Probably shouldn’t change this...

File: README.md [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+[![GoDoc](https://godoc.org/github.com/gorilla/feeds?status.svg)](https://godoc.org/github.com/gorilla/feeds)
 
 feeds is a web feed generator library for generating RSS, Atom and JSON feeds from Go
 applications.
 
+This is a fork by gmemstr which aims to make feeds more "podcast-centered". Essentially making
```
Comment (kisielk) [inline README.md]: Should be removed

File: atom.go [hunk 3]
```diff
@@ -110,19 +110,22 @@ func newAtomEntry(i *Item) *AtomEntry {
...
 		Updated: anyTimeFormat(time.RFC3339, i.Updated, i.Created),
 	}
 
-	intLength, err := strconv.ParseInt(i.Link.Length, 10, 64)
-
-	if err == nil && (intLength > 0 || i.Link.Type != "") {
-		i.Link.Rel = "enclosure"
-		x.Link = &AtomLink{Href: i.Link.Href, Rel: i.Link.Rel, Type: i.Link.Type, Length: i.Link.Length}
+	if i.Enclosure != nil && link_rel != "enclosure" {
+		// if intLength, err := strconv.ParseInt(i.Enclosure.Length, 10, 64); err == nil && intLength > 0 {
```
Comment (kisielk) [inline atom.go]: Commented or code should be removed

File: rss.go [hunk 4]
```diff
@@ -98,11 +97,13 @@ func newRssItem(i *Item) *RssItem {
 		item.Source = i.Source.Href
 	}
 
-	intLength, err := strconv.ParseInt(i.Link.Length, 10, 64)
-
-	if err == nil && (intLength > 0 || i.Link.Type != "") {
-		item.Enclosure = &RssEnclosure{Url: i.Link.Href, Type: i.Link.Type, Length: i.Link.Length}
+	// Define a closure
+	if i.Enclosure != nil && i.Enclosure.Type != "" && i.Enclosure.Length != "" {
+		// if intLength, err := strconv.ParseInt(i.Enclosure.Length, 10, 64); err == nil && intLength > 0 {
```
Comment (kisielk) [inline rss.go]: Remove commented out code

**Comment by gmemstr** [comment]
will do shortly - just headee home now. didn't actually expect to make this PR but thanks to florentheobin for the fixed.

**Comment by kisielk** [comment]
Thanks.

This changes the API, but I think if we tag a release before, we can probably merge this without feeling too guilty.

@elithrar any thoughts on this one?

**Comment by elithrar** [comment]
Agree RE: tagging. If we tag a major version before / after it should be
OK.

**Comment by kisielk** [comment]
I'm ok with it going ahead. @elithrar should we tag 1.0 for the current version of the code, and then 1.1 after merging this? though that doesn't really give people an opportunity to lock to 1.0 in the meantime so not sure if it's worth it since their code will break anyway. Maybe we should just do the bandaid pull approach and merge this and then tag 1.0?

**Comment by elithrar** [comment]
Let’s do that - just tag 1.0 after this. Neither approach is ideal, but I
agree that few people are probably locking to a version as-is.

Will get better in future!

**Comment by elithrar** [comment]
v1.0.0 tagged: https://github.com/gorilla/feeds/releases/tag/v1.0.0

**Comment by elithrar** [comment]
Thanks @gmemstr for contributing this!

**Pull request merged**